test/lora_core_tests.erl

```diff
@@ -1,12 +1,150 @@
 -module(lora_core_tests).
 %% ==================================================================
-%% Tests
+%% EUNIT Tests
 %% ==================================================================
 -ifdef(EUNIT).
 
 -include("lora.hrl").
 -include_lib("eunit/include/eunit.hrl").
 
+payload_util_test() ->
+    ValidHex00 = is_hex_string(<<"a0">>),
+    ?assertEqual(true, ValidHex00),
+    {Pay0, _Key0, _AppKey} = sample_downlink(),
+    ValidHex0 = is_hex_string(Pay0),
+    ?assertEqual(true, ValidHex0),
+    {Sample0, _, _} = sample_00(),
+    ValidHex1 = is_hex_string(Sample0),
+    ?assertEqual(false, ValidHex1),
+    Bin0 = string_to_binary(Pay0),
+    io:format("bin0 = ~w~n", [Bin0]),
+    Bin1 = string_to_binary(Sample0),
+    io:format("bin1 = ~w~n", [Bin1]),
+    {Pay2, _Key2, _AppKey2} = join_accept_sample_2(),
+    ValidHex2 = is_hex_string(Pay2),
+    ?assertEqual(true, ValidHex2),
+    Bin2 = string_to_binary(Pay2),
+    io:format("bin2 = ~w~n", [Bin2]),
+    fin.
+
+payload_decode_test() ->
+    {Pay0, _Key0, _AppKey} = sample_downlink(),
+    decode_payload(Pay0),
+    fin.
+
+payload_00_test() ->
+    decode_encode(fun sample_downlink/0),
+    fin.
+
+payload_01_test() ->
+    decode_encode(fun sample_uplink/0),
+    fin.
+
+payload_02_test() ->
+    decode_encode(fun sample_uplink_2/0),
+    fin.
+
+payload_03_test() ->
+    decode_encode(fun sample_join_request_01/0),
+    fin.
+
+payload_04_test() ->
+    decode_encode(fun join_accept_sample_2/0),
+    fin.
+
+payload_05_test() ->
+    decode_encode(fun sample_02/0),
+    fin.
+
+payload_06_test() ->
+    decode_encode(fun sample_03/0),
+    fin.
+
+% payload_07_test() ->
+%     decode_encode(fun sample_join_request_02/0),
+%     fin.
+
+payload_08_test() ->
```

Comment:
Since each call to 'encode_decode' is independent of the other, I wonder if a better structure is to generate a list of tests using all of the sample funs to call 'decode_encode' on them. That would be denser, but could partially lose the grouping by type. May be more effort than it's worth.